app/v2/route.js

```diff
@@ -0,0 +1,26 @@
+import Route from '@ember/routing/route';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+import { inject as service } from '@ember/service';
+import { jwtDecode } from 'jwt-decode';
+import { tracked } from '@glimmer/tracking';
+import { get } from '@ember/object';
+
+export default class NewRoute extends Route.extend(AuthenticatedRouteMixin) {
+  @service session;
+
+  @service router;
+
+  @tracked token = this.session.get('data.authenticated.token');
+
+  get isAdmin() {
+    const token = get(this, 'token');
+
+    return (jwtDecode(token).scope || []).includes('admin');
+  }
+
+  beforeModel() {
+    if (!this.isAdmin) {
+      this.router.transitionTo('/404');
+    }
+  }
+}
```

Comment:
`AuthenticatedRouteMixin` is deprecated.  You should refactor this to:
```suggestion
import Route from '@ember/routing/route';
import { service } from '@ember/service';

export default class NewRoute extends Route {
  @service session;

  @service router;

  beforeModel(transition) {
    this.session.requireAuthentication(transition, 'login');
    
    const isAdmin = (this.session.data.authenticated.scope || []).includes('admin');
    if (!isAdmin) {
      this.router.transitionTo('/404');
    }
  }
}
```

Reply:
Also note, all child routes are now protected by the session.